src/packet/userid.js

```diff
@@ -87,15 +87,36 @@ class UserIDPacket {
      * The `name` and `comment` parts can include any letters, whitespace, and symbols, except for `(` and `)`,
      * since they interfere with `comment` parsing.
      */
-    const re = /^(?<name>[^()]+\s+)?(?<comment>\([^()]+\)\s+)?(?<email><\S+@\S+>)$/;
-    const matches = re.exec(userID);
-    if (matches !== null) {
-      const { name, comment, email } = matches.groups;
-      this.comment = comment?.replace(/^\(|\)|\s$/g, '').trim() || ''; // remove parenthesis and separating whiltespace
-      this.name = name?.trim() || '';
-      this.email = email.substring(1, email.length - 1); // remove brackets
-    } else if (/^[^\s@]+@[^\s@]+$/.test(userID)) { // unbracketed email: enforce single @ and no whitespace
-      this.email = userID;
+
+    const isValidEmail = str => /^[a-zA-Z0-9._%+-]+@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z0-9]{2,63}$/.test(str);
```

Comment:
We had some issues with weird email addresses in the past; to minimize breaking changes let's keep the old regex for email address validation for now:
```suggestion
    const isValidEmail = str => /^[^\s@]+@[^\s@]+$/.test(str); // enforce single @ and no whitespace
```

Reply:
Thank you, I've applied the suggested change for now.